### ISSUE TITLE
Cannot remove all files & click done to save

### ISSUE THREAD

**Issue by fabien** [uploadcare/uploadcare-widget#329]
In the MultiWidget, it's possible to delete individual files. However, when deleting all of them, the 'Done' button will be disabled, and there's no way to save the wanted 'empty' state (in other words, leave the FileGroup/value empty). When closing the dialog using the (X) this is not persisted, and the state is reverted.

**Comment by dmitry-mukhin** [comment]
interesting...

**Comment by homm** [comment]
You are looking for [cleareable](https://uploadcare.com/documentation/widget/#option-clearable) option.
<img width="316" alt="screen shot 2016-03-31 at 17 08 41" src="https://cloud.githubusercontent.com/assets/128982/14178421/a2865f32-f76b-11e5-93fc-f9b40cfc3e92.png">

**Comment by fabien** [comment]
Not really, because this will remove all items at once. You *could* use this option as a workaround, in order to clear the last remaining file, but it's counter-intuitive. Just try to delete files from a multi upload widget and you'll see what I mean.

**Comment by homm** [comment]
> Not really, because this will remove all items at once.

This is what you are looking for, isn't it?

**Comment by fabien** [comment]
Yes & no ;-) while editing the previously uploaded list (reordering etc.) you might decide to remove some files, perhaps even all of them. If you decided this upfront, then `clearable` would do exactly this, however, should you decide during arbitrary edits, it becomes a two-step process.

**Comment by homm** [comment]
Well, I partly agree with you. The user may decide to save no files. But at the moment the dialog has very clear meaning: it can be "_resolved with files_" or "_rejected_" by the user. And widget always treats the dialog rejecting as "user does not change its choice".

So maybe we need to implement new state "_dialog is resolved with no files_", but this will be confusing for all API users and at least, should be done in next major version.

**Comment by fabien** [comment]
Sounds good, it's not really rejecting, and to make this clear, the 'done' button should remain enabled even if there are no files left. Perhaps you can introduce a config option for backwards-compatibility?

Could you re-open this issue, please?
